{"pageProps":{"questions":[{"id":"3lNWMo1J37sugQHnEau0","unix_timestamp":1742487360,"question_text":"What is the expected output of the following code?\n//IMG//","timestamp":"2025-03-20 17:16:00","answer_images":[],"answers_community":["E (100%)"],"choices":{"C":"1 1","D":"3 1","A":"1 42","B":"The code is erroneous.","E":"3 42"},"isMC":true,"answer":"E","answer_description":"","discussion":[{"timestamp":"1742813520.0","upvote_count":"1","content":"Selected Answer: E\nE. 3 42\nExplanation:\np1 = 1 just reassigns the local copy of p1. It does not affect the original variable.\np2[0] = 42 modifies the list in-place, which does affect the original list outside the function.\nx = 3\ny = [1, 2, 3]\nfunc(x, y):\n p1 gets a copy of x → p1 = 3, but p1 = 1 does not affect x\n p2 is a reference to the same list as y, so p2[0] = 42 changes y[0]\nx is still 3\ny[0] is now 42","poster":"Donny_575","comment_id":"1409636"},{"upvote_count":"1","poster":"akumo","content":"Selected Answer: E\nE. 3 42","timestamp":"1742487360.0","comment_id":"1401182"}],"url":"https://www.examtopics.com/discussions/python-institute/view/169532-exam-pcep-30-02-topic-4-question-12-discussion/","question_id":196,"exam_id":194,"topic":"4","question_images":["https://img.examtopics.com/pcep-30-02/image336.png"],"answer_ET":"E"},{"id":"lrX9BonCNXjFIylWENx5","topic":"4","answers_community":["AC (100%)"],"question_id":197,"answer_ET":"AC","timestamp":"2025-03-20 17:20:00","answer_images":[],"answer":"AC","choices":{"C":"fun(0, 1, 2)","B":"fun(b=1)","D":"fun()","A":"fun(b=0, a=0)"},"question_text":"Which of the following lines correctly invoke the function defined below: (Choose two.)\n//IMG//","unix_timestamp":1742487600,"exam_id":194,"discussion":[{"poster":"Donny_575","comment_id":"1409637","timestamp":"1742813940.0","upvote_count":"1","content":"Selected Answer: AC\nA and C\nExplanation: C has a default value assigned (0), so only a and b require arguments, c is optional. fun(b=0, a=0) provides the 2 required arguments and fun(0, 1, 2) provides the 2 required and 1 optional argument."},{"poster":"akumo","upvote_count":"1","content":"Selected Answer: AC\nA and C","comment_id":"1401184","timestamp":"1742487600.0"}],"url":"https://www.examtopics.com/discussions/python-institute/view/169533-exam-pcep-30-02-topic-4-question-13-discussion/","isMC":true,"question_images":["https://img.examtopics.com/pcep-30-02/image338.png"],"answer_description":""},{"id":"CyktaOUGfWGUbVpCiVTO","exam_id":194,"choices":{"A":"4","C":"The code is erroneous.","B":"5","D":"3"},"unix_timestamp":1742487780,"question_images":["https://img.examtopics.com/pcep-30-02/image340.png"],"answer_ET":"A","answer":"A","timestamp":"2025-03-20 17:23:00","question_text":"What is the output of the following snippet?\n//IMG//","answers_community":["A (100%)"],"topic":"4","isMC":true,"url":"https://www.examtopics.com/discussions/python-institute/view/169534-exam-pcep-30-02-topic-4-question-14-discussion/","question_id":198,"answer_description":"","discussion":[{"poster":"akumo","timestamp":"1742487780.0","comment_id":"1401186","content":"Selected Answer: A\nA. 4 checked","upvote_count":"1"}],"answer_images":[]},{"id":"eCISTMMglKTM3JAwnW0C","answer":"E","choices":{"C":"print 6","B":"cause a runtime exception on Line 8","E":"print 8","D":"print 4","A":"cause a runtime exception on Line 9"},"answers_community":["E (100%)"],"exam_id":194,"isMC":true,"question_images":["https://img.examtopics.com/pcep-30-02/image344.png"],"topic":"4","answer_ET":"E","unix_timestamp":1742488140,"discussion":[{"poster":"Donny_575","timestamp":"1742891760.0","content":"Selected Answer: E\nE.\nExplanation: \nLine 1: x = 1\n>>>Sets x to 1.\nLine 4–5: Defines function a(x)\n>>The function a(x) takes a value and returns 2 * x.\nLine 8: x = 2 + a(x)\n>>> At this point, x is 1 (from Line 1). So a(x) becomes a(1) → 2 * 1 = 2.\n>>> Now x = 2 + 2 = 4. So now x = 4.\nLine 9: print(a(x))\n>>> Now we call a(4) → 2 * 4 = 8. So the output is 8","comment_id":"1409949","upvote_count":"1"},{"timestamp":"1742488140.0","content":"Selected Answer: E\nThe output is 8 because after the value of x is updated to 4, the function a(x) is called with x = 4, and it returns 8.","comment_id":"1401189","upvote_count":"1","poster":"akumo"}],"question_id":199,"timestamp":"2025-03-20 17:29:00","answer_description":"","answer_images":[],"question_text":"What is the expected behavior of the following snippet?\n//IMG//\n\nIt will:","url":"https://www.examtopics.com/discussions/python-institute/view/169536-exam-pcep-30-02-topic-4-question-16-discussion/"},{"id":"rMV1FyZ0NWBzRqKETIrM","answer_ET":"D","answer_images":[],"choices":{"D":"Defines a function, which does nothing.","E":"Defines an empty class.","C":"None of the above.","A":"Defines a list and initializes it.","B":"Defines a function, which passes its parameters through."},"answers_community":["D (100%)"],"answer":"D","isMC":true,"discussion":[{"comment_id":"1401191","poster":"akumo","upvote_count":"2","timestamp":"1742488320.0","content":"Selected Answer: D\nD. Defines a function, which does nothing."}],"answer_description":"","url":"https://www.examtopics.com/discussions/python-institute/view/169537-exam-pcep-30-02-topic-4-question-17-discussion/","question_id":200,"exam_id":194,"question_images":["https://img.examtopics.com/pcep-30-02/image346.png"],"timestamp":"2025-03-20 17:32:00","question_text":"What does the following code do?\n//IMG//","topic":"4","unix_timestamp":1742488320}],"exam":{"numberOfQuestions":330,"name":"PCEP-30-02","provider":"Python-institute","isImplemented":true,"lastUpdated":"12 Apr 2025","isMCOnly":true,"id":194,"isBeta":false},"currentPage":40},"__N_SSP":true}